lib/dialects/mssql/data-types.js

```diff
@@ -153,6 +153,10 @@ var FLOAT = function() {
 };
 util.inherits(FLOAT, BaseTypes.FLOAT);
 
+BaseTypes.ENUM.prototype.toSql = function() {
+  return 'VARCHAR(10)';
```

Comment:
There some code for this in mssql query-generator. https://github.com/sequelize/sequelize/blob/master/lib/dialects/mssql/query-generator.js#L353-L361

I think it fits a lot better here, so could you move the rest of the code into this